### PR TITLE
feat: update cli to retrieve promotion commits for product release steps

### PR DESCRIPTION
**Pull request by scme0**
## What
Adding promotion commits to product release steps if the version of codefresh is compatible. I've added a fall back in case the cli is connected to an old version of codefresh (on prem).

## Why
A customer wanted to know the commit created by promotions for each step of a product release for each application. See Jira ticket for details.

## Notes
Jira: https://codefresh-io.atlassian.net/browse/CR-25720

### CODE REVIEW

File: cmd/commands/product-release.go [hunk 1]
```diff
@@ -126,9 +158,15 @@ query getProductReleasesList(
 		},
 	}
 
-	productReleasesPage, err := client.GraphqlAPI[productReleaseSlice](ctx, cfConfig.NewClient().InternalClient(), query, variables)
+	productReleasesPage, err := client.GraphqlAPI[productReleaseSlice](ctx, cfConfig.NewClient().InternalClient(), latest_query, variables)
 	if err != nil {
-		return fmt.Errorf("failed to get product releases: %w", err)
+		if strings.Contains(err.Error(), "Cannot query field \\\"promotionCommits\\\" on type \\\"ProductReleaseStep\\\".") {
+			log.G().Warn("codefresh version v0.13.4 or older detected. Using v0.13.4 query which excludes promotionCommits.")
```
Comment (scme0) [inline cmd/commands/product-release.go]: I added this warning, but now I'm not sure as if there are customers/internal users using this cli command in automation the extra log line of output would make it fail.

I can also just make it fallback silently?

File: cmd/commands/product-release.go [hunk 2]
```diff
@@ -37,6 +38,55 @@ type (
...
+				}
+			}
+			status
+			}
+		}
+	}
+}
+`
+
+const v0_13_4_query = `
```
Comment (scme0) [inline cmd/commands/product-release.go]: I'll update the specific version references once I've merged the argo-platform changes.

File: cmd/commands/product-release.go (hunk 1 above)
Comment (ATGardner) [reply cmd/commands/product-release.go]: i think that should be ok, showing a warning, but falling back to previous query.
this way they will know that something may work better if they upgrade their platform.

**Comment by scme0** [comment]
I've tested this against g.codefresh while it runs a pre v1.3120.1 version and it correctly logs a warning and displays the results without the applications property.
I've also tested it against sandbox running a branch build of v1.3120.1 and it successfully returns the applications property.

**Review by ATGardner — APPROVED** [review]

**Pull request merged**